### PR TITLE
fix(logs): LogRetention Custom Resource does not use Node18

### PR DESCRIPTION
**Pull request by mrgrain**
Migrate the LogRetention Custom Resource to AWS SDK v3.

Part of a fix for #26461 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by mrgrain** [comment]
Added exemptions as existing tests provide coverage.

**Review by rix0rrr — APPROVED** [review]

File: packages/aws-cdk-lib/aws-logs/lib/log-retention-provider/index.ts [hunk 1]
```diff
@@ -206,19 +174,41 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
...
+
+  return async (block: () => Promise<void>) => {
+    do {
+      try {
+        return await block();
+      } catch (error: any) {
+        if (error instanceof Logs.OperationAbortedException || error.name === 'OperationAbortedException') {
+          if (maxRetries > 0) {
+            maxRetries--;
+            await new Promise(resolve => setTimeout(resolve, delay));
```
Comment (rix0rrr) [inline packages/aws-cdk-lib/aws-logs/lib/log-retention-provider/index.ts:203]: Can we  make this exponential delay?

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/aws-cdk-lib/aws-logs/lib/log-retention-provider/index.ts (hunk 1 above)
Comment (rix0rrr) [reply packages/aws-cdk-lib/aws-logs/lib/log-retention-provider/index.ts:203]:
```
delay *= 2;
```

Comment (mrgrain) [reply packages/aws-cdk-lib/aws-logs/lib/log-retention-provider/index.ts:203]: Going with 10s as a max now. This creates this distribution for the default of 10 rounds, for a total of ~20s.
```
[
    60,    70,   90,
   130,   210,  370,
   690,  1330, 2610,
  5170, 10000
]
20730ms
```

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**